### PR TITLE
Move webhook type from int to string

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

File: models/migrations/v160.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		DISCORD
+		DINGTALK
+		TELEGRAM
+		MSTEAMS
+		FEISHU
+		MATRIX
+	)
+
+	var hookTaskTypes = map[int]string{
+		GITEA:    "gitea",
```
Comment (6543) [inline models/migrations/v160.go:25]: gitea is 3 in const above

File: modules/notification/webhook/webhook.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 	"code.gitea.io/gitea/modules/repository"
 	"code.gitea.io/gitea/modules/setting"
 	api "code.gitea.io/gitea/modules/structs"
-	webhook_module "code.gitea.io/gitea/modules/webhook"
+	webhook_module "code.gitea.io/gitea/services/webhook"
```
Comment (6543) [inline modules/notification/webhook/webhook.go]: I think we should rename _module to _service ...

**Comment by 6543** [comment]
@lunny if you merge https://github.com/lunny/gitea/pull/11 and resolve migration conflict it should be done

**Comment by lunny** [comment]
@6543 done.

File: services/webhook/webhook.go [hunk 3]
```diff
@@ -20,6 +20,55 @@ import (
 	"github.com/gobwas/glob"
 )
 
+type webhook struct {
+	name           models.HookTaskType
```
Comment (6543) [inline services/webhook/webhook.go:24]:
```
golangci-lint run --timeout 5m
services/webhook/webhook.go:24:2: `name` is unused (structcheck)
	name           models.HookTaskType
	^
make: *** [Makefile:703: golangci-lint] Error 1
```

Comment (lunny) [reply services/webhook/webhook.go:24]: Done.

**Review by 6543 — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Comment by jolheiser** [comment]
This migration needs to be updated to convert the webhook model as well.  
I'm not sure if more code needs to be changed as well from a cursory glance.

https://github.com/go-gitea/gitea/blob/abb9cffe4a9b36c124b1ef8fad4a00eff4ba36de/models/webhook.go#L115

**Pull request merged**